rq/worker.py

```diff
@@ -790,17 +792,25 @@ def monitor_work_horse(self, job, queue):
             if not job.ended_at:
                 job.ended_at = utcnow()
 
-            # Unhandled failure: move the job to the failed queue
-            self.log.warning((
-                'Moving job to FailedJobRegistry '
-                '(work-horse terminated unexpectedly; waitpid returned {})'
-            ).format(ret_val))
-
-            self.handle_job_failure(
-                job, queue=queue,
-                exc_string="Work-horse was terminated unexpectedly "
-                           "(waitpid returned %s)" % ret_val
-            )
+            if job.is_stopped:
```

Comment:
Instead of putting this check in this `if` block, would be cleaner and easier to understand if we do
```python
if job_status == JobStatus.STOPPED:
    ...
elif job_status not in [JobStatus.FINISHED, JobStatus.FAILED]:  # originally line 790
    ...

```